auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -135,22 +218,44 @@
      * @param root the root path of the identity store
      * @param nameRewriter the name rewriter to apply to looked up names
      * @param levels the number of levels of directory hashing to apply
-     * @param encoded whether identity names should be BASE32 encoded before using as filename
+     * @param encoded whether identity names should be BASE32 encoded before using as filename (only applies if the security realm is unencrypted)
      * @param hashCharset the character set to use when converting password strings to a byte array. Uses UTF-8 by default.
      * @param hashEncoding the string format for the hashed passwords. Uses Base64 by default.
+     * @param secretKey the SecretKey used to encrypt and decrypt the security realm (if {@code null}, the security realm will be unencrypted)
+     * @param providers The providers supplier
```

Comment:
privateKey and publicKey should be added here too.

Just minor, but secretKey should also be moved under providers.

Reply:
Added and moved